# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/base/Document.cpp
# Commit: d19dbdb939af
# Full Hash: d19dbdb939af89443317fc7a2b26c64f44569d56
# Author: Brendan Dahl <bdahl@mozilla.com>
# Date: 2019-05-21 04:19:40
# Regressor Bug: 1551320
# File Overlap Count: 1
# Description:
#   Bug 1551320 - Don't allow CreateElement in XUL documents. r=bzbarsky
#   
#   Soon XUL documents will be loaded as XHTML and createElement will create HTML
#   elements instead of XUL element by default. This restriction can be removed
#   once we migrate everything.
# ==============================================================================

diff -r 307d9921bdf2 -r d19dbdb939af dom/base/Document.cpp
--- a/dom/base/Document.cpp	Mon May 20 15:38:48 2019 +0000
+++ b/dom/base/Document.cpp	Thu May 16 07:12:15 2019 +0000
@@ -5483,6 +5483,14 @@
     return nullptr;
   }
 
+  // Temporary check until XULDocument has been removed.
+  if (IsXULDocument()) {
+    MOZ_DIAGNOSTIC_ASSERT(false,
+                          "CreateElement() not allowed in XUL document.");
+    rv.Throw(NS_ERROR_FAILURE);
+    return nullptr;
+  }
+
   bool needsLowercase = IsHTMLDocument() && !IsLowercaseASCII(aTagName);
   nsAutoString lcTagName;
   if (needsLowercase) {
